### PR TITLE
chore: Cristi/increase coverage

### PR DESCRIPTION
**Pull request by CristiCanizales**
### What does this PR do?
- Increases tests coverage

### Related work item
@W-12226851@

### CODE REVIEW

File: packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts [hunk 1]
```diff
@@ -14,7 +17,16 @@ jest.mock('@salesforce/core', () => {
...
+  const FAKE_WORKSPACE_INSTANCE: any = {};
   const FAKE_STATE_FOLDER = '.sfdx';
+  const FAKE_METADATA = 'metadata';
+  const FAKE_ORGS = 'orgs';
+  const FAKE_TEST_RESULTS = 'testresults';
+  const FAKE_APEX = 'apex';
+  const FAKE_APEX_DB = 'apex.db';
+  const FAKE_DEBUG = 'debug';
+  const FAKE_LOGS = 'logs';
+  const FAKE_LWC = 'lwc';
```
Comment (gbockus-sf) [inline packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts:29]: Rather then duplicate these here we should export the const values from paths.ts

Comment (CristiCanizales) [reply packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts:29]: yess. I was going to ask you about it - to do the same we're doing with tools. Thank you!

File: packages/salesforcedx-utils-vscode/test/jest/context/workspaceContextUtil.test.ts [hunk 2]
```diff
@@ -150,5 +155,22 @@ describe('WorkspaceContext', () => {
...
+    it('should throw error if there is no username set', async () => {
+      getUsernameStub.mockReturnValue(undefined);
+
+      await mockWatcher.onDidChange.mock.calls[0][0]();
+
+      try {
+        await workspaceContextUtil.getConnection();
+      } catch (error) {
+        expect(error).toEqual(new Error(nls.localize('error_no_default_username')));
+      }
```
Comment (gbockus-sf) [inline packages/salesforcedx-utils-vscode/test/jest/context/workspaceContextUtil.test.ts:173]: You can use the following syntax here and avoid the issue where if the getConnection call doesn't throw the test still passes
```
      expect(() => {
        await workspaceContextUtil.getConnection();
      }).toThrowError(nls.localize('error_no_default_username'));
```

File: packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts [hunk 3]
```diff
@@ -78,4 +81,162 @@ describe('test project paths', () => {
...
+    });
+
+    it('should return a path to the tools folder based on root workspace', () => {
+      const toolsFolder = projectPaths.toolsFolder();
+      expect(toolsFolder).toEqual(path.join(FAKE_WORKSPACE, TOOLS));
+    });
+  });
+
+  describe('test metadataFolder', () => {
+    jest.setTimeout(10000);
```
Comment (gbockus-sf) [inline packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts:104]: why is this required?

File: packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts [hunk 4]
```diff
@@ -78,4 +81,162 @@ describe('test project paths', () => {
...
+      usernameStub = jest
+      .spyOn(WorkspaceContextUtil, 'getInstance')
+      .mockReturnValue(FAKE_WORKSPACE_INSTANCE);
+    });
+    it('should be defined', () => {
+      expect(projectPaths.metadataFolder).toBeDefined();
+    });
+
+    it('should return a path to the metadata folder based on root workspace', () => {
+      // expect(usernameStub).toHaveBeenCalled();
```
Comment (gbockus-sf) [inline packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts:122]: nit: no dead code.

File: packages/salesforcedx-utils-vscode/test/unit/auth/authUtil.test.ts [hunk 5]
```diff
@@ -7,8 +7,9 @@
 
 import { StateAggregator } from '@salesforce/core';
 import { expect } from 'chai';
-import { createSandbox, SinonSandbox, stub } from 'sinon';
-import { AuthUtil } from '../../../src';
```
Comment (gbockus-sf) [inline packages/salesforcedx-utils-vscode/test/unit/auth/authUtil.test.ts:11]: This is fine for this PR, but in the future don't bother increasing coverage on unit tests and write a jest test for the file instead.

**Review by gbockus-sf — APPROVED** [review]
cool thanks 👍 a few nits to address and it's good to merge.

File: packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts (hunk 3 above)
Comment (CristiCanizales) [reply packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts:104]: Tests were failing without it, but now apparently it's been fixed automagically 🤡

File: packages/salesforcedx-utils-vscode/test/jest/context/workspaceContextUtil.test.ts (hunk 2 above)
Comment (CristiCanizales) [reply packages/salesforcedx-utils-vscode/test/jest/context/workspaceContextUtil.test.ts:173]: mm that doesn't seem to be working for me🥲

File: packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts [hunk 6]
```diff
@@ -78,4 +81,162 @@ describe('test project paths', () => {
...
+      expect(projectPaths.debugLogsFolder).toBeDefined();
+    });
+
+    it('should return a path to the debug Logs folder based on root workspace', () => {
+      const debugLogsFolder = projectPaths.debugLogsFolder();
+      expect(debugLogsFolder).toEqual(path.join(FAKE_WORKSPACE, TOOLS, DEBUG, LOGS));
+    });
+  });
+
+  describe('test fileExtensionsMatch IN PROGRESS', () => {
```
Comment (CristiCanizales) [inline packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts:219]: @gbockus-sf also need some help with these... Haven't been able to make them pass🥲

Comment (gbockus-sf) [reply packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts:217]: I'll push up a commit

File: packages/salesforcedx-utils-vscode/test/jest/context/workspaceContextUtil.test.ts (hunk 2 above)
Comment (gbockus-sf) [reply packages/salesforcedx-utils-vscode/test/jest/context/workspaceContextUtil.test.ts:173]: example pushed in commit

File: packages/salesforcedx-utils-vscode/test/jest/context/workspaceContextUtil.test.ts [hunk 7]
```diff
@@ -127,28 +143,48 @@ describe('WorkspaceContext', () => {
...
       expect(connection).toEqual(mockConnection);
     });
 
     it('should return a cached connection for the default org if there is one', async () => {
       authInfoMock.create.mockResolvedValue(mockAuthInfo as any);
-      connectionMock.create.mockResolvedValue(mockConnection as unknown as Promise<Connection<any>>);
+      connectionMock.create.mockResolvedValue(
+        (mockConnection as unknown) as Promise<Connection<any>>
+      );
       await workspaceContextUtil.getConnection();
```
Comment (gbockus-sf) [inline packages/salesforcedx-utils-vscode/test/jest/context/workspaceContextUtil.test.ts:167]: my formatter went a little nuts.  We should check if you have prettier running locally in vscode.

File: packages/salesforcedx-utils-vscode/test/jest/context/workspaceContextUtil.test.ts [hunk 8]
```diff
@@ -127,28 +143,48 @@ describe('WorkspaceContext', () => {
...
+    it('should throw error if there is no username set', async () => {
+      getUsernameStub.mockReturnValue(undefined);
+
+      await mockWatcher.onDidChange.mock.calls[0][0]();
+
+      const message = nls.localize('error_no_default_username');
+      // tslint:disable-next-line:no-floating-promises
+      expect(async () => {
+        await workspaceContextUtil.getConnection();
+      }).rejects.toThrowError(message);
```
Comment (gbockus-sf) [inline packages/salesforcedx-utils-vscode/test/jest/context/workspaceContextUtil.test.ts:187]: Every time I do this I have to explore a little to find the right combo to get the expect to work.  Also the tslint exclusion is lame, but it's about time to pull the trigger on just moving everything to eslint so I'm ok with it.

File: packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts [hunk 9]
```diff
@@ -78,4 +98,176 @@ describe('test project paths', () => {
...
+      expect(debugLogsFolder).toEqual(
+        path.join(FAKE_WORKSPACE, TOOLS, DEBUG, LOGS)
+      );
+    });
+  });
+
+  describe('test fileExtensionsMatch IN PROGRESS', () => {
+    const mockEditor = {
+      document: {
+        uri: { path: 'foo.log' } as vscode.Uri
```
Comment (gbockus-sf) [inline packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts:249]: Ok so the tricky part here is you have to keep in mind that all things vscode.* in jest tests are mocked.  So you were calling a jest mock when you called vscode.Uri.file and it was just resolving nothing.

File: packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts [hunk 10]
```diff
@@ -78,4 +98,176 @@ describe('test project paths', () => {
...
+      }
+    };
+
+    beforeEach(() => {
+      (vscode.window.activeTextEditor as any).mockReturnValue(mockEditor);
```
Comment (gbockus-sf) [inline packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts:254]: Had to add activeTextEditor to the setup-jest.ts config file so it existing for mocking.

File: scripts/setup-jest.ts [hunk 11]
```diff
@@ -53,6 +53,7 @@ const getMockVSCode = () => {
       Right: 2
     },
     window: {
+      activeTextEditor: jest.fn(),
```
Comment (gbockus-sf) [inline scripts/setup-jest.ts:56]: This is just getting built out as we go.

**Comment by gbockus-sf** [comment]
Note the PR validate action is failing b/c you don't have a WI in your PR description

**Comment by gbockus-sf** [comment]
I created a ticket for this work and added it to the description.  We may need to close and reopen the PR to get the pr validation check to pass.

**Comment by gbockus-sf** [comment]
I'm gonna try closing and reopening this PR to see if the check will clear. Otherwise we'll just bypass

**Review by gbockus-sf — APPROVED** [review]
